day-01/day-01-main.cc

```diff
@@ -1,15 +1,59 @@
 // Advent of Code 2020
 
-#include <iostream>
+#include <unordered_map>
 
-#include "../helper-classes/list-reader.h"
+#include "../helper-classes/int-list-reader.h"
+
+std::unique_ptr<std::pair<int, int>> Get2Sum(int target_value,
+        std::vector<int> value_list, int used_value = INT_MIN) {
+    // build the complement map
+    std::unordered_map<int, int> complement_map;
```

Comment:
Instead of a compliment map, why not just have the values directly in an `unordered_set`?

E.g.

```c++
    std::unordered_set values (value_list.begin(), value_list.end());

   for (int value : values) {
      const int complement = target_value - value;
      if (values.find(complement) != values.end() {
           return std::make_pair(value, complement);
      }
  }

    // this function should not be used for values that don't have a twosum.
    assert();
```
http://www.cplusplus.com/reference/unordered_set/unordered_set/find/

Reply:
I end up using the stored value in the pair, both for 2sum but especially for 3sum. Instead of just storing it in a separate variable, I figured I might as well keep it associated with the value it is used with.